benfoster/peekaqueue
# GTW-1106: Support Retry Policy To SQS

### Jira link
-  [GTW-1106](https://checkout.atlassian.net/browse/GTW-1106)

### Technical comments
- SqsConnectionRetryCount setting added -> retry connection count
- SqsConnectionRetryBackoffMultiplier setting added -> backoff multiplier

### Functional comments
peekaqueue should be resilient to network issues when is trying to connect to SQS. Retry policy is implemented such that if an exception is thrown during the GetQueueURL it retries. 

### How to test
- Turn off the alpine-sqs
- Run peekaqueue
- On the second retry turn on alpine-sqs
- Check the console for the retry logs 

[Click here for the PR help checklist](https://checkout.atlassian.net/wiki/spaces/GW/pages/385876274/PR+checklist)
